python/tank/platform/bundle.py

```diff
@@ -804,28 +790,143 @@ def __post_process_settings_r(self, key, value, schema):
         
         return processed_val
         
-    def __resolve_setting_value(self, key, value):
+    def __resolve_setting_value(self, settings, key, default):
         """
-        Resolve a setting value.  Exposed to allow values
-        to be resolved for settings derived outside of the 
-        app.
-        
-        :param key:   setting name
-        :param value: setting value
+        Resolve a setting value.  Exposed to allow values to be resolved for
+        settings derived outside of the app.
+
+        :param settings: the settings dictionary source
+        :param key: setting name
+        :param default: a default value to use for the setting
+        """
+
+        # The post processing code requires the schema to introspect the
+        # setting's types, defaults, etc. An old use case exists whereby the key
+        # does not exist in the config schema so we need to account for that.
+        schema = self.__descriptor.get_configuration_schema().get(key, None)
+
+        # Get the value for the supplied key
+        if key in settings:
+            # Value provided by the settings
+            value = settings[key]
+        elif schema:
+            # Resolve a default value from the schema. This checks various
+            # legacy default value forms in the schema keys.
+            value = resolve_default_value(schema, default,
+                self._get_engine_name())
+        else:
+            # Nothing in the settings, no schema, fallback to the supplied
+            # default value
+            value = default
+
+        # We have a value of some kind and a schema. Allow the post
+        # processing code to further resolve the value.
+        if value and schema:
+            value = self.__post_process_settings_r(key, value, schema)
+
+        return value
+
+    def _get_engine_name(self):
+        """Returns the bundle's engine name if available. None otherwise.
+
+        Convenience method to avoid try/except everywhere.
+
+        :return: The engine name or None
         """
-        if value is None:
-            return value
-        
-        # try to get the type for the setting
-        # (may fail if the key does not exist in the schema,
-        # which is an old use case we need to support now...)
+
         try:
-            schema = self.__descriptor.get_configuration_schema().get(key)
+            engine_name = self.engine.name
```

Comment:
feels like we should have note here (like in the code that was remoted) to highlight the fact that this is the `bundle` class and not all bundle instances have a `self.engine` - this is the reason we have to put that try/catch around it